### PR TITLE
Add PlayerFlowerPotManipulateEvent

### PR DESCRIPTION
**Pull request by MisterVector**
Added an event for when a player places an item in or takes an item out of a flower pot.

This is a revisit to an old PR I had submitted nearly 3 years ago but wasn't able to finish because of an issue with server-side code that handles event cancellation.

### CODE REVIEW

File: Spigot-Server-Patches/0445-Implement-PlayerFlowerPotManipulateEvent.patch [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
++            org.bukkit.block.Block bukkitblock = org.bukkit.craftbukkit.block.CraftBlock.at(world, blockposition);
++            org.bukkit.inventory.ItemStack bukkititemstack = org.bukkit.craftbukkit.inventory.CraftItemStack.asBukkitCopy(itemstack);
++            org.bukkit.Material mat = org.bukkit.craftbukkit.util.CraftMagicNumbers.getMaterial(c);
++            org.bukkit.inventory.ItemStack bukkititemstack1 = new org.bukkit.inventory.ItemStack(mat, 1);
++            org.bukkit.inventory.ItemStack whichitem = placing ? bukkititemstack : bukkititemstack1;
++
++            com.destroystokyo.paper.event.player.PlayerFlowerPotManipulateEvent event = new com.destroystokyo.paper.event.player.PlayerFlowerPotManipulateEvent(player, bukkitblock, whichitem, placing);
++            player.getServer().getPluginManager().callEvent(event);
++
++            if (event.isCancelled()) {
```
Comment (jacobsandersen) [inline Spigot-Server-Patches/0445-Implement-PlayerFlowerPotManipulateEvent.patch:30]: Just test the result of callEvent, as it handles event cancellation already.

**Review by jacobsandersen — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0445-Implement-PlayerFlowerPotManipulateEvent.patch [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+         Block block = item instanceof ItemBlock ? (Block) BlockFlowerPot.b.getOrDefault(((ItemBlock) item).getBlock(), Blocks.AIR) : Blocks.AIR;
+         boolean flag = block == Blocks.AIR;
+         boolean flag1 = this.c == Blocks.AIR;
+-
++ 
+         if (flag != flag1) {
++            // Paper start
++            org.bukkit.entity.Player player = (org.bukkit.entity.Player) entityhuman.getBukkitEntity();
++            boolean placing = flag1;
++            org.bukkit.block.Block bukkitblock = org.bukkit.craftbukkit.block.CraftBlock.at(world, blockposition);
```
Comment (jacobsandersen) [inline Spigot-Server-Patches/0445-Implement-PlayerFlowerPotManipulateEvent.patch:21]: Starting here, this is extremely verbose with fully-qualified classes. Try using imports when you are using the same class more than once.

**Review by jacobsandersen — CHANGES_REQUESTED** [review]

**Comment by kickash32** [comment]
I suggest you use imports in BlockFlowerPot.java as currently the implementation of the event is very hard to read.

**Comment by kickash32** [comment]
Additionally master is not accepting new feature PRs, only the 1.14 branch is. You need to port this patch to the 1.14 branch and then submit a new PR.

**Comment by MisterVector** [comment]
> 
> 
> Additionally master is not accepting new feature PRs, only the 1.14 branch is. You need to port this patch to the 1.14 branch and then submit a new PR.

According to https://papermc.io/forums/t/july-project-status-update/1391 they aren't formally feature-freezing the master branch (Minecraft 1.13.2). So on that note what should I do?

**Comment by kickash32** [comment]
Although its not documented here, it has been talked about on the discord and PRs will only be accepted for 1.13 if they are bug fixes or also PR'd for 1.14.

**Comment by MisterVector** [comment]
Alright then that is what I will do.

**Pull request closed without merging**